src/commands/emulators-start.ts

```diff
@@ -106,6 +107,12 @@ Issues? Report them at ${stylizeLink(
     )} and attach the *-debug.log files.
  `);
 
+    // Log the command for analytics, we report the emulator as "hub"
+    // since we originally mistakenly reported emulators:start events
+    // for each emulator, by reporting the "hub" we ensure that our
+    // historical data can still be viewed.
+    track("emulators:start", "hub");
```

Comment:
Aren't there some situations where hub does not run? Maybe we should just track this in the very top level of the command action.

Reply:
Is there? I thought it was a pre-req for any emulator?

Reply:
Pretty sure the hub only runs if we detect a Project ID. @yuchenshi

Reply:
So hub runs...

```js
shouldStart() {
  if (name === Emulators.HUB) {
    // The hub only starts if we know the project ID.
    return !!options.project;
  }
  ...
}
  ```
  
 So I guess that could mean the hub doesn't always start?

Reply:
Yep `!!options.project` is checking for a project ID.

Reply:
If that's the case, we could switch to just reporting `emulators:start` with no other args, but that'll make it hard to correlate to historical data. At least if we report "hub" we can filter that in GA and get close.

Reply:
Let's just take the `track()` out of `startEmulator` and put it in this block:

```ts
  if (shouldStart(options, Emulators.HUB)) {
    const hubAddr = await getAndCheckAddress(Emulators.HUB, options);
    const hub = new EmulatorHub({ projectId, ...hubAddr });
    
    track("emulators:start", "hub"); // ADD THIS LINE
    await startEmulator(hub);
  }
  ```
  
  This way the `emulators:start, hub` line in Analytics means the same thing tomorrow as it does today.